source/howto/remote-access/index.rst

```diff
@@ -0,0 +1,153 @@
+
+.. meta::
+    :description: How remotely access and debug live execution
+
+.. _executions-compare:
+
+Remote access (SSH)
+################################
+
+Valohai offers remote access to a live execution with `SSH <https://en.wikipedia.org/wiki/Secure_Shell>`_ (Secure Shell).
+It is a low-level and agnostic protocol, which makes it usable for a wide array of tasks.
+
+**Common use-cases**
+
+* Inspecting the execution using an interactive terminal
+* Connect your favorite IDE debugger like VSCode or PyCharm
+* Low-latency integration with 3rd party tools like Tensorboard
+
+.. admonition:: Prerequirements
+  :class: attention
+
+  * TODO
+
+1. Create an SSH keypair
+----------------------------------
+
+An SSH keypair is required for securing the connection. You may re-use an existing keypair, but please be mindful of
+regenerating it periodically according to your security standards.
+
+Use :code:`ssh-keygen` to create a new SSH key pair.
+
+.. code-block:: bash
+
+   $ ssh-keygen -t rsa -b 4096 -N '' -f my-debug-key
+
+This will generate two files:
+
+* :code:`my-debug-key.pub` is the public key you paste into UI before starting an execution.
+* :code:`my-debug-key` is the private key you need for connecting to the execution.
+
+.. admonition:: Don't include the keys in your version control
+   :class: warning
+
+   You should **not** include these keys in the version control. Anybody that gains access to the :code:`valohai-debug-key` file contents will have access to your execution, so use appropriate caution.
+..
+
+2. Start an execution
+----------------------------------
+
+Valohai GUI
+===========
+
+.. thumbnail:: /howto/remote-access/start-execution.png
+   :alt: Start Valohai execution with SSH enabled
+..
+
+Start a Valohai execution with the "Run with SSH" enabled.
+
+Copy-paste the entire contents of the :code:`my-debug-key.pub` file into the text field. Change the TCP/IP port if
+your network setup requires it.
+
+Valohai CLI
+===========
+
+Start a Valohai execution with extra parameter :code:`debug-key-file` for your public key file (and additionally :code:`debug-port`).
+
+.. code-block:: bash
+
+   vh exec run -a --debug-key-file=/tmp/remote-debug-key.pub train
```

Comment:
I suggest we use the full name here, just so it's a tad easier to understand for new users. You don't need to guess what does `-a` stand for (instead you get to guess what does adhoc mean :D)

```suggestion
   vh exec run --adhoc --debug-key-file=/tmp/remote-debug-key.pub train
```